### PR TITLE
set the mutatingwebhookconfiguration generation after create to avoid…

### PR DESCRIPTION
**Pull request by ashleyschuett**
… an unnecessary update

Signed-off-by: Ashley Schuett <aschuett@redhat.com>

**What this PR does / why we need it**:

I introduced this bug in my recent reconciliation PR. We should update the KV status with the generation on create so that we don't patch the object when it is not needed. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-operator/resource/apply/admissionregistration.go [hunk 1]
```diff
@@ -172,12 +172,14 @@ func (r *Reconciler) createOrUpdateMutatingWebhookConfiguration(webhook *admissi
...
 		r.expectations.MutatingWebhook.RaiseExpectations(r.kvKey, 1, 0)
-		_, err := r.clientset.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().Create(context.Background(), webhook, metav1.CreateOptions{})
+		webhook, err = r.clientset.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().Create(context.Background(), webhook, metav1.CreateOptions{})
 		if err != nil {
 			r.expectations.MutatingWebhook.LowerExpectations(r.kvKey, 1, 0)
 			return fmt.Errorf("unable to create mutatingwebhook %+v: %v", webhook, err)
 		}
 
+		SetMutatingWebhookConfigurationGeneration(&r.kv.Status.Generations, webhook)
+
```
Comment (dhiller) [inline pkg/virt-operator/resource/apply/admissionregistration.go:182]: Maybe worth adding a log statement here?

```suggestion
log.Log.V(2).Infof("mutatingwebhoookconfiguration %v created", webhook.Name)
```

**Review by dhiller — APPROVED** [review]
/approve
/hold 

Up to you to decide if you want to follow my suggestion, feel free to unhold in that case.

**Comment by ashleyschuett** [comment]
@dhiller  thank you for the suggestion! the log should definitely be there. 

/unhold

**Pull request merged**